### ISSUE TITLE
Star Wars - Knights of the Old Republic: Game save corruption

### ISSUE THREAD

**Issue by Shoegzer** [xemu-project/xemu#1409]
### Title

https://xemu.app/titles/4c410003/#Star-Wars-Knights-of-the-Old-Republic

### Bug Description

After playing the game for 15-30 minutes, the game will attempt to auto-save.  Sometimes this works, but often it doesn't.  This results in a corrupt game save that cannot be resolved from that point forward.  Of course, this renders the game unplayable.

Steps to reproduce:
1. Start a game, create a character and play for 30 mins or so
2. Leave the main ship via Escape Pod
3. Explore the planet below for a few minutes.
4. Observe game attempting to auto-save and failing, resulting in "Damaged Game" appearing whenever loading is attempted:
![pic](https://user-images.githubusercontent.com/12475371/226146544-5f801132-feee-4824-ad26-5095fe4051fe.png)

### Expected Behavior

Games should save without corruption

### xemu Version

Version:      0.7.85
Branch:       master
Commit:       065c74a00fe402f65147e9b0d1a1eeac1b4b609f
Date:         Tue Mar 14 06:34:06 PM UTC 2023 

### System Information

CPU:          Intel(R) Core(TM) i7-10870H CPU @ 2.20GHz
OS Platform:  Linux
OS Version:   Linux Mint 21.1
Manufacturer: NVIDIA Corporation
GPU Model:    NVIDIA GeForce RTX 3070 Laptop GPU/PCIe/SSE2
Driver:       4.0.0 NVIDIA 525.85.05
Shader:       4.00 NVIDIA via Cg compiler

### Additional Context

I made a copy of the hard drive and tried playing the game multiple times, but each time the save became corrupted at roughly the same time in the game as noted above.

Also, this game's status should be changed from "Playable" to "Starts" on the compatibility page.

**Comment by Shoegzer** [comment]
Note: Others have experienced this issue as well - see the comments on [this](https://www.youtube.com/watch?v=I7hHMNrDSiU) YouTube video.  Interestingly, the video's creator has not experienced the problem.  Curious if the issue is platform-specific such as OS.

**Comment by ShalokShalom** [comment]
This seems like a rather drastic issue. IDK about the tradition around labeling with Xemu, but is this still worth calling the game playable at the time?

**Comment by Shoegzer** [comment]
Agreed.  I would certainly not say it's playable.  I gave up on trying to finish it.

**Comment by ShalokShalom** [comment]
Who can change the status?

**Comment by dracc** [comment]
> Note: Others have experienced this issue as well - see the comments on [this](https://www.youtube.com/watch?v=I7hHMNrDSiU) YouTube video. Interestingly, the video's creator has not experienced the problem. Curious if the issue is platform-specific such as OS.

@Shoegzer Sounds more to me like a game dump specific problem. The CPU and PATA bus are quite well understood, and the main players in save handling for xemu. File system specifics are handled by the BIOS dump the user provides.

Does your game dump work on your physical Xbox?

@ShalokShalom 
![image](https://github.com/xemu-project/xemu/assets/1279635/69f05f7c-0733-4803-adef-d35563cac65e)

**Comment by Shoegzer** [comment]
The dump tested does work on an xbox.  It's not a dump issue..

**Comment by ShalokShalom** [comment]
@dracc Thanks a lot. I am not using Discord. 
May you be interested, to report the new status?

**Comment by JerryBerry117** [comment]
This issue still persists, I just got save corruption issue. Game should be relabeled as it this behavior makes it not playable, 

xemu Version
Version: 0.7.118
Branch: master
Commit: https://github.com/xemu-project/xemu/commit/03f40b1d8e873b57eab68dc66ae9892aa5e39f89
Date: Tue Mar 10 16:19:06 PM UTC 2024

OS Platform: Windows
OS Version: Windows 10

**Comment by Moeinich** [comment]
I am also having this issue..

I am on a steam deck OLED

**Comment by mlwhan2209** [comment]
got damaged save issue in the apartments in lower city, about hour and a half in. shouldnt be marked as playable anymore really
